File: Desafio Mobile iOS/Desafio Mobile iOS/Models/PullRequest.m
//
//  PullRequest.m
//  Desafio Mobile iOS
//
//  Created by Adriano Rezena on 12/10/17.
//  Copyright © 2017 Adriano Rezena. All rights reserved.
//

#import "PullRequest.h"

@implementation PullRequest

- (id)initWithDictionary:(NSDictionary*)dictionary andRepositoryId:(NSInteger)repositoryId {
    if (self = [super init]) {
        
        self.id = (long)dictionary[@"id"];
        self.repositoryId = repositoryId;
        self.url = dictionary[@"html_url"];
        self.state = dictionary[@"state"];
        self.title = dictionary[@"title"];
        self.body = dictionary[@"body"];
        
        NSDateFormatter *dateFormatter = [NSDateFormatter new];
        [dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ssZ"];
    
        self.created_at = [dateFormatter dateFromString:dictionary[@"created_at"]];
        
        self.owner = [[Owner alloc] initWithDictionary:dictionary[@"user"]];
    }
    
    return self;
}


+ (NSString *)primaryKey {
    return @"id";
}


@end

File: GitHubAllStars/GitHubAllStars/Data/Models/PullRequest.swift
//
//  PullRequest.swift
//  GitHubAllStars
//
//  Created by Raphael Araújo on 23/10/17.
//  Copyright © 2017 Raphael Araújo. All rights reserved.
//

import UIKit
import RealmSwift
import ObjectMapper
class PullRequest: Object, Mappable {
    @objc dynamic var id: Int32 = 0
    @objc dynamic var title: String = ""
    @objc dynamic var body: String = ""
    @objc dynamic var user: Owner?
    @objc dynamic var repository: Repository?
    @objc dynamic var created_at: String = ""
    @objc dynamic var html_url: String = ""
    
    override class func primaryKey() -> String? {
        return "id"
    }
    
    required convenience init?(map: Map) {
        self.init()
    }
    
    func mapping(map: Map) {
        id <- map["id"]
        title <- map["title"]
        body <- map["body"]
        user <- map["user"]
        created_at <- map["created_at"]
        html_url <- map["html_url"]
        repository <- map["base.repo"]
    }
}
